lib/ansible/module_utils/aws/rds.py

```diff
@@ -0,0 +1,125 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible. If not, see <http://www.gnu.org/licenses/>.
+
+import botocore
+
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict
+
+
+def get_db_instance(conn, instancename):
+    try:
+        response = conn.describe_db_instances(DBInstanceIdentifier=instancename)
+        instance = RDSDBInstance(response['DBInstances'][0])
+        return instance
+    except botocore.exceptions.ClientError as e:
+        if e.response['Error']['Code'] == 'DBInstanceNotFound':
+            return None
+        else:
+            raise
+
+
+def get_db_snapshot(conn, snapshotid):
+    try:
+        response = conn.describe_db_snapshots(DBSnapshotIdentifier=snapshotid)
+        snapshot = RDSSnapshot(response['DBSnapshots'][0])
+        return snapshot
+    except botocore.exceptions.ClientError as e:
+        if e.response['Error']['Code'] == 'DBSnapshotNotFound':
+            return None
+        else:
+            raise
+
+
+class RDSDBInstance(object):
+    def __init__(self, dbinstance):
+        self.instance = dbinstance
+        if 'DBInstanceIdentifier' not in dbinstance:
+            self.name = None
+        else:
+            self.name = self.instance.get('DBInstanceIdentifier')
+        self.status = self.instance.get('DBInstanceStatus')
+        self.data = self.get_data()
+
+    def get_data(self):
+        return camel_dict_to_snake_dict(self.instance)
+
+    def diff(self, params):
+        # FIXME compare_keys should be all things that can be modified
+        # except port and instance_name which are handled separately
+        # valid_vars = ['backup_retention', 'backup_window',
+        #               'db_name',  'db_engine', 'engine_version',
+        #               'instance_type', 'iops', 'license_model',
+        #               'maint_window', 'multi_zone', 'new_instance_name',
+        #               'option_group', 'parameter_group', 'password', 'size',
+        #               'storage_type', 'subnet', 'tags', 'upgrade', 'username',
+        #               'vpc_security_groups']
+        compare_keys = ['backup_retention', 'instance_type', 'iops',
+                        'maintenance_window', 'multi_zone',
+                        'replication_source',
+                        'size', 'storage_type', 'tags', 'zone']
+        leave_if_null = ['maintenance_window', 'backup_retention']
+        before = dict()
+        after = dict()
+        for k in compare_keys:
+            if self.data.get(k) != params.get(k):
+                if params.get(k) is None and k in leave_if_null:
+                    pass
+                else:
+                    before[k] = self.data.get(k)
+                    after[k] = params.get(k)
+        try:
+            old_port = self.data.get("endpoint")["port"]
+        except TypeError:
+            old_port = None
+        try:
+            new_port = params.get("endpoint")["port"]
```

Comment:
Same suggestion as above. .get() doesn't have to return a Nonetype so you can use it for "port" too.